0004/README.md

```diff
@@ -0,0 +1,524 @@
+result in the installation of a rogue ‘foo' package version that was built from
+an unexpected repository or with an unexpected builder. 
+
+If a ‘good' package version does not meet policy expectations, the expectations
+need to be updated for that package version and onwards, so that earlier package
+versions will continue to pass expectations (different dependencies may require
+different versions of a package).
+
+These expectations can only protect the parts of the software supply chain that
+follow after after a provenance attestation has been signed, for example, they can
+protect against a tampered package in a distribution repository. 
+
+The expectations cannot protect against 'upstream' compromises, for example the
+corruption of a build system, the exposure of a signing key, or a malicious
+commit to an expected source repository. 
+
+### Expectations when publishing a package to a registry
+
+A developer submits a new version of a package to a registry. A policy evaluator
+at the registry verifies the signature on the SLSA provenance for that package,
+and verifies that the provenance meets the provenance expectations for this
+version of the package (one policy covers multiple versions of a specific
+package).
+
+Initially the policy evaluator will run pre-upload or post-upload, producing
+alerts only, with expectations stored at the evaluator. For enforcement the
+evaluator will need to be integrated into the package registry, and expectations
+may become managed centrally.
+
+A developer may also run a policy evaluator locally, with their own provenance
+expectations, before attempting to submit their package to a registry.
+
+An organization that uses a private registry may have different expectations and
+verification than a public registry.
+
+### Expectations when installing a package from a registry
+
+A developer installs a software package as an application or library dependency,
+or an end user installs an application package in a user directory. In both
+cases the npm package manager installs transitive dependencies for the specified
+package, and records each installed package in the local lock file
+(package-lock.json).
+
+A policy evaluator at the developer or end user applies a specific SLSA
+provenance policy for each package version that is recorded in the local lock
+file  (one policy covers multiple versions of a specific package).
+
+Initially the policy evaluator will run post-installation, producing alerts
+only, using expectations stored at the local evaluator instance. For enforcement
+the evaluator will need to be integrated into the package installer, and
+expectations may become managed centrally. 
+
+### Generating provenance expectations
+
+To overcome a bootstrapping problem, we assume that compromises will be rare,
+and propose to generate package-specific SLSA provenance expectations on first
+use. Each such ‘trust on first use' policy contains expectations based on
```

Comment:
I'm not sure TOFU policies will scale beyond a certain point. How should package developers, package registries, and package users be expected to keep updating their policies when things break?

Reply:
Indeed, TOFU is only a for bootstrapping, not a long-term, full-scale solution. If I need to make that more explicit, then I will do that.

Reply:
SGTM. Following my earlier comment, could we please mark distribution as out of scope for this proposal, and explicitly state that TOFU is not a long-term solution? Thanks!

Reply:
Added under  'non-objectives/policy distribution' that the TOFU approach supports prototyping policy and policy evaluation, and that it is not an alternative for secure management and distribution.